decoder.go

```diff
@@ -274,19 +279,26 @@ func (d *decoder) decode(nalu []byte) (*avFrameWrapper, error) {
 	if dst.frame.width != d.src.width || dst.frame.height != d.src.height {
 		d.logger.Debugf("(re)making frame due to AVFrame dimension discrepancy: Dst (width: %d, height: %d) vs Src (width: %d, height: %d)",
 			dst.frame.width, dst.frame.height, d.src.width, d.src.height)
+
+		// Handle size changes while having previously initialized frames to avoid https://github.com/erh/viamrtsp/pull/41#discussion_r1719998891
+		frameWasPreviouslyInitialized := dst.frame.width > 0 && dst.frame.height > 0
+		if frameWasPreviouslyInitialized {
+			// Release previously initialized frames, and block old prev gen frames from returning to pool
+			dst.free()
+			d.avFramePool.clearAndStartNewGeneration()
+			// Make new frame to be initialized with new size
+			newDst, err := newAVFrameWrapper()
```

Comment:
IIUC, this `newDst.generation` is going to be `-1`.

I think that object will ultimately be returned to the pool in the future, but I don't see us ever updating its generation to the right value? Are we really not seeing [this panic](https://github.com/viam-modules/viamrtsp/pull/50/files#diff-5c0127b47636d901a18590597909d34f79c1e67840557675b5a981c26f7601e2R77)? I'm hoping I'm just missing something simple here.

Reply:
you're right

just did some manual testing

sorry for sending this out for review before manual testing